src/Folder.php

```diff
@@ -237,7 +237,8 @@ public function onAfterWrite()
         parent::onAfterWrite();
 
         // No publishing UX for folders, so just cascade changes live
-        if (class_exists(Versioned::class) && Versioned::get_stage() === Versioned::DRAFT) {
+        if (class_exists(Versioned::class) && File::has_extension(Versioned::class) && Versioned::get_stage() === Versioned::DRAFT)
```

Comment:
Should we rather use `$this->hasExtension`?

Reply:
right, that's not really gonna work here since it has different semantics. I guess static would be more flexible then?

Reply:
though, maybe that's what we actually need here? @maxime-rainville what do you think?

Reply:
I think it makes some degree of sense to specifically look at whether Files has extensions applied to it rather than Folder. While Folder extends File, there's bits of logic that tries to suppress their Versioned behaviour. In a perfect world Folder wouldn't extends File.